erpnext/accounts/doctype/journal_entry/journal_entry.py

```diff
@@ -687,26 +687,39 @@ def get_payment_entry(ref_doc, args):
 		"remark": args.get("remarks")
 	})
 
-	party_row = je.append("accounts", {
-		"account": args.get("party_account"),
-		"party_type": args.get("party_type"),
-		"party": ref_doc.get(args.get("party_type").lower()),
-		"cost_center": cost_center,
-		"account_type": frappe.db.get_value("Account", args.get("party_account"), "account_type"),
-		"account_currency": args.get("party_account_currency") or \
-			get_account_currency(args.get("party_account")),
-		"balance": get_balance_on(args.get("party_account")),
-		"party_balance": get_balance_on(party=args.get("party"), party_type=args.get("party_type")),
-		"exchange_rate": exchange_rate,
-		args.get("amount_field_party"): args.get("amount"),
-		"is_advance": args.get("is_advance"),
-		"reference_type": ref_doc.doctype,
-		"reference_name": ref_doc.name
-	})
+	options_ref_name_list = [
+		d.due_date.strftime(DATE_FORMAT) for d in ref_doc.payment_schedule
+		if d.get('due_date')
+	]
+
+	for payment_term in ref_doc.payment_schedule:
```

Comment:
What if no payment_schedule?

Reply:
Will work. I thought we migrated Sales/Purchase orders to automatically create Payment Schedule